### DISCUSSION TITLE
Overlayed (Non Summing) Stacked Bar Chart

### DISCUSSION THREAD

**Discussion by EP-PCOM** [recharts/recharts#4926] [Ideas]
It would be wonderful if there was an option to simply overlay the data within a stacked bar chart, rather than to simply summarize it, similar to the image included here. The value being hovered over is what highlights/displays.

<img width="1175" alt="Screen Shot 2024-08-26 at 12 10 24 PM" src="https://github.com/user-attachments/assets/3aea66a8-5f88-4c86-9ff5-29eafd10a8ff">

**Comment by ckifer** [comment]
Are you asking about the split, fixed tooltip? Or about the data where "62" is showing as the value from 0 to 62 instead of from 25 to 87?

**Reply by EP-PCOM** [reply]
I'm talking about how the two sets of data (calls and converted) overlap one another rather than stack on top as an aggregate. Current versions with those two datasets provided appear to sum and result in the total value on the chart being 87, rather than the highest point between the two sets ie 62.

**Reply by ckifer** [reply]
I'm not quite sure that a stacked Bar chart is the chart type you want if you need values to represent themselves rather than to represent a secondary category that totals to a whole. Why not just a regular bar chart with multiple series?

Taking a look at some guidelines such as this https://www.atlassian.com/data/charts/stacked-bar-chart-complete-guide most sources seem to reflect that stance.

If you look at other charting libraries they default to the same behavior https://www.chartjs.org/docs/latest/samples/bar/stacked.html

However, I totally understand that requirements are requirements. This _might_ possible with `Customized`, but [Customized will be deprecated in 3.0](https://github.com/recharts/recharts/discussions/4437) as things get pretty hacky with it. I gave it a quick attempt but wasn't able to get it working, Bar doesn't provide as much useful info in Customized as Line and Scatter do (see website example https://recharts.org/en-US/examples/CustomizedRectangleContent)

**Comment by draymondxie** [comment]
I have the same requirement too, as recharts didn't provide the overlay feature of bar chart, how did you solve that in the end?

**Comment by PavelVanecek** [comment]
If you put the Bars on different XAxis then it renders overlapped: https://main--63da8268a0da9970db6992aa.chromatic.com/?path=/story/api-chart-barchart--bar-in-bar

<img width="963" height="436" alt="image" src="https://github.com/user-attachments/assets/7fd367b6-afe2-4293-9b81-b5ffc1f08f75" />

**Reply by draymondxie** [reply]
Hi Pavel, thanks for your advice, wanna know if the mutiple XAxis souliton is fit for the composed chart too? In my use case, my composed chart will have stacked bars and lines, and they shared the same XAxis currently. Appreciate your feedback, thanks!

**Reply by draymondxie** [reply]
By the way, I have found a temp solution to meet my requirement, by defining the bar size of each bars, and set up the bar gap property  to the negative value of the defined bar size, seems it works so far:
<img width="1627" height="718" alt="image" src="https://github.com/user-attachments/assets/8d0cad7f-a95e-494f-9e50-8edd1f42cb7d" />
